x-pack/plugins/security_solution/server/endpoint/lib/policy/license_watch.ts

```diff
@@ -91,21 +91,22 @@ export class PolicyWatcher {
         return;
       }
       response.items.forEach(async (policy) => {
-        const policyConfig = policy.inputs[0].config?.policy.value;
+        const { id, ...updatePolicy } = policy;
```

Comment:
I don't think this is going to do it. The type for [`UpdatePackagePolicy`](https://github.com/elastic/kibana/blob/284b1046df6981f0da1e743680bf99247523627e/x-pack/plugins/fleet/common/types/models/package_policy.ts#L48-L61) (Fleet) does not have several of the properties  defined in the `PackagePolicy` which we get from the list API _(I'm surprised its taking in the `revision` - wondering if that is doing anything at all when the update occurs)_ 

My suggestion would be to do something similar to what we do in https://github.com/elastic/kibana/blob/80ca5a5836e537c2695231b38249a4499d80be5b/x-pack/plugins/security_solution/public/management/pages/policy/store/policy_details/selectors.ts#L33

Perhaps the above method I reference should be elevated to `security_solution/common/` and then re-used in both frontend and server?

_(ps. I realize now that even the method I referenced can still "leak" properties)

Reply:
so, yes, you're right in that there are extra properties not being stripped in order to match `UpdatePackagePolicy` perfectly. I don't know why typescript isn't complaining more about this.

that being said, only `id` is getting rejected during save. The rest are ignored. 

as for destructuring further, we'd be starting a sisyphus task. Any time a field might be added to PackagePolicy, and not explicitly added to UpdatePackagePolicy, it would need to be destructured here too.

We can aid ourselves a little by combining the two places it may happen as you mention.

But I would say no, no need here. My reasoning being:

1. This does fix the bug behavior. Everything succeeds here
2. So doing more destructuring and cleanup would be a refactor / improvement. But not a very clean or future-proof one. This can be cleaned up, but that doesn't seem any more robust. We should actually do that robustly (perhaps explicitly creating a new UpdatePolicyPackage object, and assigning the properties explicitly. So the default is to drop extra props))

Reply:
😄 
Ok. Maybe you can get a 👍 from Fleet then? Maybe @jfsiii or @neptunian or @nchaulet ?

(a bit more - some of which I mentioned in our slack conversation)

The problem I see _(and I will likely suggest changes to Fleet services if possible)_ is that for fleet, this is not an issue because they use the service from the API and have Schema’s in place to validate the input - so no extra properties are going to make it in. We however are using it via their service which means we have to ensure the data is valid. We're fortunate in this case in that the update [service overrides the extra properties we're passing in](https://github.com/elastic/kibana/blob/4e4e55029ce05cee5a7642df7bfbaaeb10f1529b/x-pack/plugins/fleet/server/services/package_policy.ts#L298-L304).

Typescript types by default are not "exact" but rather if an object with x amount of properties has enough of them to full-fill the targe Type, then typescript is happy, which is the case here. Really (IMO) the types for the input arguments to the service need to be defined as `Exact<>` _(note: this is not supported in Typescript today, but a proposal)_. There are ways to do this in Typescript today until the utility type lands ([see here](https://github.com/elastic/kibana/blob/fb48e903d5506f0a9dd235fb494c463eef8e2d70/x-pack/plugins/security_solution/public/management/common/routing.ts#L26-L28))

Reply:
I think @pzl has a good point in that we'd have to maintain destructuring - however the one part that scares me is that it doesn't look like `created_at` or `created_by` are overridden, so we'd presumably be overwriting that.

the code I'm looking at: https://github.com/elastic/kibana/blob/4e4e55029ce05cee5a7642df7bfbaaeb10f1529b/x-pack/plugins/fleet/server/services/package_policy.ts#L298-L304

Agreed with Dan's point that the types should be stricter and we should just create a new `UpdatePackagePolicy` object that enforces that we're only passing the expected fields